### PR TITLE
Add gormigrate migration helper

### PR DESCRIPTION
**Pull request by alexpalyan**

### CODE REVIEW

File: pkg/orm/orm.go [hunk 1]
```diff
@@ -54,12 +58,26 @@ func (c *OrmConfig) setDefaults(
...
+	// ID is the migration identifier. Usually a timestamp like "201601021504".
+	ID string
+	// Migrate is a function that will br executed while running this migration.
+	Migrate gormigrate.MigrateFunc
+	// Rollback will be executed on rollback. Can be nil.
+	Rollback gormigrate.RollbackFunc
+}
+
+// NewMySqlOrm - creates a new Orm object with MySQL connection
+// TODO: rename to NewMySQLOrm
```
Comment (mattiasavelin) [inline pkg/orm/orm.go]: Why not right away? 😁

File: pkg/orm/orm.go [hunk 2]
```diff
@@ -54,12 +58,26 @@ func (c *OrmConfig) setDefaults(
...
 type Orm struct {
 	*gorm.DB
+	config *OrmConfig
 }
 
+// Migration represents a database migration (a modification to be made on the database).
+type Migration struct {
+	// ID is the migration identifier. Usually a timestamp like "201601021504".
+	ID string
+	// Migrate is a function that will br executed while running this migration.
```
Comment (mattiasavelin) [inline pkg/orm/orm.go]: Typo? br -> be

File: pkg/orm/orm.go (hunk 1 above)
Comment (alexpalyan) [reply pkg/orm/orm.go]: Because it may be used in other projects... I decided not to risk...

File: pkg/orm/orm.go (hunk 2 above)
Comment (alexpalyan) [reply pkg/orm/orm.go]: Yes :) Fixed :)

**Comment by andreas-lindfalk** [comment]
I changed this a bit, see the quite extensive migration-section in the Readme for how these two migration options live side by side and one can choose which way to use.

The reason for the changes I've made is that Gormigrate was sort of wrapped behind our own definitions, but not all of the options was possible to change... and also, it just adds complexity/layers/mapping to wrap the lib. So now it's more "plain vanilla Gormigrate" :)

File: pkg/orm/orm.go [hunk 3]
```diff
@@ -64,18 +61,7 @@ type Orm struct {
...
-	ID string
-	// Migrate is a function that will be executed while running this migration.
-	Migrate gormigrate.MigrateFunc
-	// Rollback will be executed on rollback. Can be nil.
-	Rollback gormigrate.RollbackFunc
-}
-
 // NewMySqlOrm - creates a new Orm object with MySQL connection
-// TODO: rename to NewMySQLOrm
 func NewMySqlOrm(config *OrmConfig) *Orm {
```
Comment (alexpalyan) [inline pkg/orm/orm.go:65]: `func NewMySQLOrm ...`
It should be renamed to be more consistent with golint rules...

File: pkg/orm/orm.go [hunk 4]
```diff
@@ -23,19 +22,17 @@ var defaultMySQLLogger = logger.Discard.LogMode(logger.Silent) // rely on Opente
 var defaultSQLiteLogger = logger.Default.LogMode(logger.Info)
 
 // OrmConfig - configuration structure for config values at ORM module
-// TODO: Rename to Config
 type OrmConfig struct {
```
Comment (alexpalyan) [inline pkg/orm/orm.go:25]: package name is "orm" and config name could be just Config :)

**Comment by alexpalyan** [comment]
> I changed this a bit, see the quite extensive migration-section in the Readme for how these two migration options live side by side and one can choose which way to use.
> 
> The reason for the changes I've made is that Gormigrate was sort of wrapped behind our own definitions, but not all of the options was possible to change... and also, it just adds complexity/layers/mapping to wrap the lib. So now it's more "plain vanilla Gormigrate" :)

Excellent! Thank you! )

File: pkg/orm/orm.go (hunk 3 above)
Comment (andreas-lindfalk) [reply pkg/orm/orm.go:65]: That would be a breaking change

File: pkg/orm/orm.go (hunk 4 above)
Comment (andreas-lindfalk) [reply pkg/orm/orm.go:25]: That would be a breaking change

**Pull request merged**